src/app/core/services/navigation/nav-tree.service.ts

```diff
@@ -12,49 +12,46 @@ export abstract class NavTreeService<ContentT extends RoutedContentInfo, Childre
 
   private reloadTrigger = new Subject<void>();
 
-  private contentInfo$ = this.currentContent.content$.pipe( // only keep those of interest for the current nav tree
+  private content$ = this.currentContent.content$.pipe( // only keep those of interest for the current nav tree
     // map those which are not of interest to `undefined`
     map(content => (this.isOfContentType(content) ? content : undefined)),
     distinctUntilChanged(), // remove multiple `undefined`
     startWith(undefined),
     repeatLatestWhen(this.reloadTrigger),
   );
-
-  state$ = combineLatest([ this.contentInfo$, this.contentInfo$.pipe(this.childrenNavigation()) ]).pipe(
-    switchScan((prevState: FetchState<NavTreeData>, [ contentInfo, children ]) => {
-
-      if (prevState.isReady) {
-        const prevData = prevState.data;
-
-        // CASE: the current content is not to be displayed in the menu and the menu has already an element displayed
-        //       -> just deselect what is selected if there was a selection
-        if (!contentInfo) {
-          return prevData.selectedElementId === undefined ? EMPTY : of(readyState(prevData.withNoSelection()));
-        }
-
-        // CASE: the content is among the displayed items
-        //       -> updates its data, and either select it if at root or shift the tree "to the left" otherwise
-        const contentId = contentInfo.route.id;
-        const contentInMenuLev1 = prevData.hasLevel1Element(contentId);
-        const contentInMenuLev2 = prevData.hasLevel2Element(contentId);
-        if (contentInMenuLev1 || contentInMenuLev2) {
-          let data = contentInMenuLev1 ? prevData.withSelection(contentId) : prevData.subNavMenuData(contentId);
-          data = data.withUpdatedElement(contentId, el => this.addDetailsToTreeElement(el, contentInfo, children));
+  private children$ = this.content$.pipe(this.childrenNavigation());
+  state$ = combineLatest([ this.content$, this.children$ ]).pipe(
+    mergeScan((prevState: FetchState<NavTreeData>, [ content, children ]) => {
+
+      // CASE 1: the content is not of type of this menu
```

Comment:
What does it mean `is not of type of this menu` ? (Same for `CASE 2`)

Reply:
nav menu can be for activities, skills or groups. How do you want me to say that without being too long?